include/libtorrent/file_storage.hpp

```diff
@@ -134,7 +134,7 @@ namespace libtorrent
 		};
 
 		// the offset of this file inside the torrent
-		std::uint64_t offset:48;
+		std::int64_t offset:48;
```

Comment:
I see, I didn't realize this was a bitfield. I agree with you then. this causes the same problem as before, it won't necessarily merge with surrounding bitfields of different types (afaik)